### DISCUSSION TITLE
issue when calling API route

### DISCUSSION THREAD

**Discussion by vladandrei0** [vercel/next.js#20848] [Help] [answered]
Hi everyone, 

I have a hook calling an API route and I get the following error:
![Capture1](https://user-images.githubusercontent.com/24414516/103883282-b0455a00-50e5-11eb-8fbf-4db3af5b83e4.PNG)

The pages structure is like this: 
/pages
 /api
 /1stfolder/[id]
 /2ndfolder[id]

If I call the hook from components inside the 1st folder it works fine, but if it gets called from the second I get the error above.
How is it possible to get this request URL in one folder: 
![image](https://user-images.githubusercontent.com/24414516/103884738-a91f4b80-50e7-11eb-81b2-b1e6efcbad49.png)
and this from the other folder: 
![image](https://user-images.githubusercontent.com/24414516/103885022-1632e100-50e8-11eb-9f9e-a54c0546cc7b.png)

I tried everything, I surrender 😣

**Comment by MehediH** [comment, marked as answer]
It looks like you are using a relative path to call the API route, could you please share how you are calling the API route from the different folders/files?

If you look at the 404 response, you can see that it is calling `localhost:3000/teme/api/..` but your API route is in `localhost:3000/api/` (like in the 201 response).

**Reply by vladandrei0** [reply]
this is the hook:
```
export default function useLikes(userId) {
   return useQuery(
      'liked',
      () => client(`api/getLiked?query=${encodeURIComponent(userId)}`)
   )
}
 ```
this is how it gets called:
```
export function RenderSheets({ fiseDinQuery, userId, status, title }) {
   if (userId) {
      const queryClient = useQueryClient()

      const { data: likedFise } = useLikes(userId)
.
.
.
```
I use the same component in both folders 🤷‍♂️

**Reply by MehediH** [reply]
Try with 

```js
client(`/api/getLiked?query=${encodeURIComponent(userId)}`)
```

instead of 

```js
client(`api/getLiked?query=${encodeURIComponent(userId)}`)
```

**Reply by vladandrei0** [reply]
It worked 🙏🙏🙏
wow!! 😂

**Reply by MehediH** [reply]
No worries ahaha, can you mark my reply as an answer so the discussion shows up as solved/answered?
